1082 group is not updated after delete user

tcnichol opened the pull request:
This is draft, I think I found a fix but my pre commit is still acting up.

tcnichol commented:
I think something is messed up here. I only wanted changes in Group component, I reverted changes in the other one, but it still shows changes.
Draft till I figure this out.

longshuicy commented:
I switched this to draft according to your comments; if this is ready feel free to switch it back. Thanks!

ddey2 commented:
@tcnichol I see this

<img width="1107" alt="Screenshot 2024-06-28 at 12 26 33 PM" src="https://github.com/clowder-framework/clowder2/assets/22405928/83f47ed3-63bb-4631-bf86-211aa158f1dc">

ddey2 commented:
@tcnichol You might want to merge main to resolve the above error. 

Also, it doesn't work. I deleted user but it doesn't update it on the page unless you refresh it.

ddey2 reviewed (changes requested):
@tcnichol  You might want to merge main to resolve the above error.

Also, it doesn't work. I deleted user but it doesn't update it on the page unless you refresh it.

tcnichol commented:
> @tcnichol You might want to merge main to resolve the above error.
> 
> Also, it doesn't work. I deleted user but it doesn't update it on the page unless you refresh it.

Can you try to delete the node_modules file and then do npm install again? When I switched to another branch and back, it wasn't working until I did npm install again.

longshuicy commented on frontend/src/components/groups/Group.tsx:
```diff
@@ -1,103 +1,82 @@
-			{/*Error Message dialogue*/}
-			<ErrorModal errorOpen={errorOpen} setErrorOpen={setErrorOpen} />
-			{/*breadcrumb*/}
-			<Grid container>
```
any reason you are removing the breadcrumb?

tcnichol replied:
No, this must have been a mistake. I have put it back in.

longshuicy commented on frontend/src/components/groups/Group.tsx:
```diff
@@ -1,103 +1,82 @@
+					groupDeleted(groupId);
+					setDeleteGroupConfirmOpen(false);
+					// Go to Explore page
+					history("/");
```
Is it ok to go to /groups page instead?

longshuicy commented on frontend/src/components/groups/Group.tsx:
```diff
@@ -149,16 +143,25 @@ export function Group() {
 						</Button>
 					</AuthWrapper>
-					<AuthWrapper currRole={role} allowedRoles={["owner", "editor"]}>
-						<EditMenu setDeleteGroupConfirmOpen={setDeleteGroupConfirmOpen} />
```
Any reason you are get rid of the edit menu which includes "edit group names etc"?

tcnichol replied:
I think this was a mistake somehow. I will fix this and check to see what else is missing from main.

longshuicy reviewed (changes requested):
This PR works but it seems to get rid of a few existing functionality. 
Also a little explanation on what's the culprit of the issue and what's the fix would be greatly appreciated!

tcnichol commented:
@longshuicy 

I went back and reverted all of these files to main, and then just made the change necessary to make this work. I'm not sure why it got changed so much.

The basic thing was the useEffect hook needed to depend on groupAbout, so it would populate the group members correctly when groupAbout changed after a user was deleted from the group. Now it updates right for delete and add, and that should be the only change.

longshuicy commented:
Thanks for the explanation. Deletion works well now.

However when I try to edit the role of any user, it wouldn't allow me to select any of those selection
<img width="561" alt="image" src="https://github.com/clowder-framework/clowder2/assets/13950475/8ba9486f-2fa7-49c3-a3a5-ddf17ee1c110">
